### ISSUE TITLE
RESTful API Endpoint - Stop Individual Stream

### ISSUE THREAD

**Issue by dbContext** [AirenSoft/OvenMediaEngine#401]
**Is your feature request related to a problem? Please describe.**
I see that you've already put the initial placeholder for the request, however it only returns a NotImplemented response.

https://github.com/AirenSoft/OvenMediaEngine/blob/master/src/projects/api_server/controllers/v1/vhosts/apps/streams/streams_controller.cpp#L109

**Describe the solution you'd like**
Send request to API, and stop/clean up resources/destroy stream based on Stream Key.

Thank you.

**Comment by getroot** [comment]
Yes this has not been implemented yet. An API to delete a stream will be implemented later.

**Comment by dbContext** [comment]
Thanks @getroot, is there a timeline on this implementation? As the stub is there, as I linked above just not complete.

If you point me in the correct direction, for methods to delete/clean up an existing stream - I can do the implementation and send a PR.

**Comment by getroot** [comment]
@dbContext 
There is no specific timeline for implementing that feature yet. This is because the OME project has many higher priority tasks scheduled. Thank you very much for trying to PR this. Even if it becomes impossible for you to implement this for various reasons, I would like to thank you very much.

Here's a little guide on how to implement this: (It's not that simple.)
When an API receives a request to delete a stream, it must distinguish whether it is an output stream or an input stream. If it's an output stream then you need to find its input stream. After that, you must command the provider that is processing the input stream to stop. Some structural changes may be necessary because a way must be implemented for external modules to command the provider's stream.

**Comment by catapop84** [comment]
Just a quick update related to this feature. It would be nice to have an event that will be triggered when a stream is closed.

You might want to know when a client stops streaming, so you can update database.

**Comment by getroot** [comment]
This feature has been released.
https://github.com/airensoft/ovenmediaengine/discussions/1072

Please use the Stream Creation API.
